src/Network/Response.php

```diff
@@ -538,13 +538,18 @@ protected function _sendHeader($name, $value = null)
 
     /**
      * Sends a content string to the client.
+     * If the content is a callable, it is invoked.
      *
      * @param string $content string to send as response body
      * @return void
      */
     protected function _sendContent($content)
     {
-        echo $content;
+        if (!is_string($content) && is_callable($content)) {
+            $content->__invoke();
```

Comment:
> Please use $content() not all callables are objects

Good call.

> Would it make sense to also echo the return value of the callable?

I don't think so, because the whole idea of a stream is that they output while you read them. If you were to return it, it would have to be put in a single variable, removing the benefit of the stream. Unless I'm missing something?